defs.mak

```diff
@@ -61,6 +61,10 @@ ifeq ($(MYST_RELEASE),1)
 OPTIMIZATION_CFLAGS += -O3
 endif
 
+ifdef MYST_ENABLE_GCOV
```

Comment:
Why do we need to enclose `GCOV_CFLAGS` in `ifdef MYST_ENABLE_GCOV`?

Reply:
We don't really need to, I kept it there to enforce this since we want to make sure that gcov flags are only used when it is enabled. Its being extra sure that future changes in the code don't accidentally use the GCOV flags, plus we had some instances in this revision as well but I am removing them